server/service/hosts.go

```diff
@@ -1770,7 +1775,7 @@ func (svc *Service) SetHostDeviceMapping(ctx context.Context, hostID uint, email
 		// Return the updated device mappings including the IDP mapping
 		return svc.ds.ListHostDeviceMapping(ctx, hostID)
 	default:
-		return nil, fleet.NewInvalidArgumentError("source", fmt.Sprintf("must be 'custom' or '%s'", fleet.DeviceMappingIDP))
+		return nil, fleet.NewInvalidArgumentError("source", fmt.Sprintf("must be 'custom' or '%s' or '%s'", DeviceMappingIDP, fleet.DeviceMappingMDMIdpAccounts))
```

Comment:
We have to update the rest-api.md docs and add `mdm_idp_accounts` as a valid value too.